docs: CORS configuration (#9773)
langflow-ai/langflow: docs/docs/Configuration/api-keys-and-authentication.mdx

Only superusers can manage user accounts for a Langflow server, but user management only matters if your server has authentication enabled.
For more information, see [Start a Langflow server with authentication enabled](#start-a-langflow-server-with-authentication-enabled).

### LANGFLOW_CORS_ORIGINS and LANGFLOW_CORS_ALLOW_CREDENTIALS {#cors-configuration-for-authentication}

Cross-Origin Resource Sharing (CORS) configuration controls how authentication credentials are handled when your Langflow frontend and backend are served from different origins.

The current default configuration allows all origins (`*`) with credentials enabled, which can be a security risk in production environments.

```text
LANGFLOW_CORS_ORIGINS=*
LANGFLOW_CORS_ALLOW_CREDENTIALS=true
```

The default settings enable the following:
- Any website can make requests to your Langflow API.
- Any website can include credentials in cross-origin requests, including authentication cookies and authorization headers.

> - API requests must include the `x-api-key` header for authentication.

Reviewer: I'm actually not sure how this third point is relevant to CORs -- I think this is a separate auth point.

Only superusers can manage user accounts for a Langflow server, but user management only matters if your server has authentication enabled.
For more information, see [Start a Langflow server with authentication enabled](#start-a-langflow-server-with-authentication-enabled).

### LANGFLOW_CORS_ORIGINS and LANGFLOW_CORS_ALLOW_CREDENTIALS {#cors-configuration-for-authentication}

Cross-Origin Resource Sharing (CORS) configuration affects how authentication credentials are handled when your Langflow frontend and backend are served from different origins (domains, ports, or protocols).

> The default settings below allow all origins (`*`) and disable credentials for security.

Reviewer 2: ```suggestion
The default settings allow all origins (`*`) and disable credentials for enhanced security:
```

```text
LANGFLOW_CORS_ORIGINS=*
LANGFLOW_CORS_ALLOW_CREDENTIALS=false
```

> With the default settings:
> - Any website can make requests to your Langflow API
> - Authentication cookies and authorization headers are not sent with requests
> - API requests must include the `x-api-key` header for authentication

Reviewer 2: ```suggestion
The default settings enable the following:

- Any website can make requests to your Langflow API.
- Authentication cookies and authorization headers aren't sent with requests.
- API requests must include the `x-api-key` header for authentication.
```

For production deployments where your frontend and backend are on different origins, configure CORS to allow specific origins and enable credentials:

```text
LANGFLOW_CORS_ORIGINS=https://yourdomain.com,https://app.yourdomain.com
LANGFLOW_CORS_ALLOW_CREDENTIALS=true
```

> With this configuration:
> - Only your specified domains can make requests to the server
> - Cookies and authorization headers are sent with requests
> - Supports session-based authentication in addition to API key authentication

Reviewer 2: ```suggestion
This configuration enables the following:

- Only your specified domains can make requests to the server.
- Cookies and authorization headers are sent with requests.
- Accepts session-based authentication and API key authentication.
```

> With CORS credentials enabled, you can use JWT authentication for your backend requests with a JWT token from your authentication service.

Reviewer 2: ```suggestion
With CORS credentials enabled, you can use JWT authentication for your backend requests with a JWT token from your authentication service:
```
